### ISSUE TITLE
particle3d render bug on real device

### ISSUE THREAD

**Issue by MarkusTieche** [cocos/cocos-engine#6277]
The trail render and probably the basic particleSystem3D has a render error on real devices. I have tested this on iPhone6 & on iPhone11, the later one seems to have render problems.

I move a cube/particleSystem in negative z direction. The camera just move in this direction as well. 

```
update (dt) 
{
      this.node.z -=10;
     this.camera.z -= 10;
},
```
Works great in simulator/web but on a real device, the trail starts to flicker and vanishes at a certain point.
(added) Does not work on mobile safari on iPhone11 as well
![renderError](https://user-images.githubusercontent.com/8083536/75790679-d4206c80-5d6b-11ea-9fe7-364ac9ee8101.gif)

**Comment by MarkusTieche** [comment]
Probably a mediump, highp problem for cc_cameraPos??

**Comment by SantyWang** [comment]
Thanks for reporting

**Comment by caryliu1999** [comment]
> Probably a mediump, highp problem for cc_cameraPos??

Can you merge this PR and see if the problem is fixed? 

https://github.com/cocos-creator/engine/pull/6280

**Comment by MarkusTieche** [comment]
The PR you mentioned did not fix the problem (if I did everything correct in building & testing).

The problem is not limited to the particleSystem3d but also to other shader elements, such as the toon shader (2.3.1 RC5). 
There seem to be something wrong with the way cc_cameraPos is calculated or transferred as shader uniform on newer iPhones. 

Toon-shader Recorded on iPhone11 in mobileSafari
![toonShaderBug](https://user-images.githubusercontent.com/8083536/75907701-0ce24380-5e49-11ea-8c56-709c9bfc7042.gif)

**Comment by MarkusTieche** [comment]
After some more research I noticed that the attribute a_position might not be correct or precise on newer iPhones. 

I changed this line in the particle shader and fixed the position for testing purposes and noticed that the flickering on iPhone 11 is gone, but now moves slightly to velocity direction. Works only with particles in local space.

https://github.com/cocos-creator/engine/blob/9428f94afbf93f2f36ce540c15740bc19e96d4ac/cocos2d/renderer/build/chunks/shader/particle-vs-legacy.inc#L25

```
 vec4 pos = vec4(vec3(.0,.0,.0), 1);
```
Tested on iPhone 11 on mobileSafari (left:modified shader)
![renderbug 2020-03-10 12_08_44](https://user-images.githubusercontent.com/8083536/76306980-2b709080-62c8-11ea-80b2-de2a5a270f9e.gif)

I really don't know why the behaviour is different on different iPhones :(. Hopefully this helps to find the problem any time soon.
Maybe transfer position separately to the shader instead of using a_position?
Or maybe it doesn't like the way this is handled
https://github.com/cocos-creator/engine/blob/9428f94afbf93f2f36ce540c15740bc19e96d4ac/cocos2d/renderer/build/chunks/shader/particle-vs-legacy.inc#L39
